pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -383,13 +387,6 @@ func (cb *ClusterBuilder) buildDefaultCluster(name string, discoveryType cluster
 		}
 	}
 
-	if discoveryType == cluster.Cluster_ORIGINAL_DST {
```

Comment:
Why remove this? Is n't this still needed for TCP upstreams?

Reply:
Please read the PR in Envoy that changed how ORIGINAL_DST works. Cleanup does not forcefully remove connections, it's just cache bookkeeping for hosts that _don't_ have connections.

Reply:
Got it Thanks